pkg/traceql/engine.go

```diff
@@ -171,7 +171,15 @@ func (e *Engine) asTraceSearchMetadata(spanset *Spanset) (*tempopb.TraceSearchMe
 			Attributes:        nil,
 		}
 
+		if name, ok := span.Attributes[NewIntrinsic(IntrinsicName)]; ok {
+			tempopbSpan.Name = name.S
+		}
```

Comment:
Currently, the name attribute will only be fetched if the query needs it. Should we always fetch it?

Reply:
I think this is ok for now.